hal/network/lwip/wiznet/wiznetif.cpp

```diff
@@ -358,16 +366,39 @@ int WizNetif::down() {
 }
 
 int WizNetif::powerUp() {
-    return 0;
+    // FIXME: As long as the interface is initialized,
+    // it must have been powered up as of right now.
+    if (pwrState_ != IF_POWER_STATE_UP) {
+        notifyPowerState(IF_POWER_STATE_UP);
+    }
+    return SYSTEM_ERROR_NONE;
 }
 
 int WizNetif::powerDown() {
-    return down();
+    int ret = down();
+    // FIXME: This don't really power off the module.
+    // Notify the system network manager that the module is powered down
+    // to bypass waitInterfaceOff() as required by system sleep.
+    if (pwrState_ != IF_POWER_STATE_DOWN) {
+        notifyPowerState(IF_POWER_STATE_DOWN);
+    }
+    return ret;
+}
+
+void WizNetif::notifyPowerState(if_power_state_t state) {
+    pwrState_ = state;
```

Comment:
This could probably be:
```c++
if (pwrState_ == state) {
    return;
}
pwrState_ = state;
// Notify
```

So that you don't have to check externally.

Reply:
That might miss a power state change event. Image that the `pwrState_ ` is changed to UP on initialization, then `powerUp()` won't notify the power state.

Reply:
Okay, let's leave it as-is then :+1:

Reply:
That's not a good example. If the `pwrState_` is changed to `IF_POWER_STATE_UP` somewhere but without notifying the state, somewhere else call `notifyPowerState(IF_POWER_STATE_UP)` won't notify the power state. Just in case.

Reply:
While this remind me that we should always notify the power state if possible, especially in `powerDown()` that may block the `waitInterfaceOff()`. I'll revise it a little bit.